xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCUser.java

```diff
@@ -589,12 +655,21 @@ private void process(
         // Packets to a specific node/group/room
         if ( preExistingRole == null )
         {
-            Log.debug("Ignoring stanza received from a non-occupant of '{}': {}", roomName, packet.toXML());
-            if ( packet.isRequest() )
+            //OF-370: check for memberlist retrieval
+            boolean allowMemberListRetrieval = checkForMemberListQuery(packet); 
+            if (allowMemberListRetrieval && packet.isRequest())
```

Comment:
This should probably be more specific than `packet.isRequest`. Instead, it should check if the request is a ListRetrieval request (which I think is now part of the implementation of `checkForMemberListQuery` - but I do not think it belongs there).

Reply:
divided into seperate methods now